mavros/src/plugins/hil.cpp

```diff
@@ -0,0 +1,356 @@
+	 * Message specification: @p https://pixhawk.ethz.ch/mavlink/#HIL_OPTICAL_FLOW
+	 */
+	void optical_flow_cb(const mavros_msgs::OpticalFlowRad::ConstPtr &req) {
+		mavlink::common::msg::HIL_OPTICAL_FLOW of;
+
+		auto int_xy = ftf::transform_frame_aircraft_baselink(
+					Eigen::Vector3d(
+						req->integrated_x,
+						req->integrated_y,
+						0.0));
+		auto int_gyro = ftf::transform_frame_aircraft_baselink(
+					Eigen::Vector3d(
+						req->integrated_xgyro,
+						req->integrated_ygyro,
+						req->integrated_zgyro));
+
+		of.time_usec = req->header.stamp.toNSec() / 1000;
+		of.sensor_id = 0;
+		of.integration_time_us = req->integration_time_us;
+		// [[[cog:
+		// for f in "xy":
+		//     cog.outl("of.integrated_%s = int_xy.%s();" % (f, f))
+		// for f in "xyz":
+		//     cog.outl("of.integrated_%sgyro = int_gyro.%s();" % (f, f))
+		// for f in ('time_delta_distance_us', 'distance', 'quality '):
+		//     cog.outl("of.%s = req->%s;" % (f, f))
+		// ]]]
+		of.integrated_x = int_xy.x();
+		of.integrated_y = int_xy.y();
+		of.integrated_xgyro = int_gyro.x();
+		of.integrated_ygyro = int_gyro.y();
+		of.integrated_zgyro = int_gyro.z();
+		of.time_delta_distance_us = req->time_delta_distance_us;
+		of.distance = req->distance;
+		of.quality  = req->quality;
+		// [[[end]]] (checksum: 7c67e6b05dad4db31ccf2872ab34fa69)
+		of.temperature = req->temperature * 100.0f;	// in centi-degrees celsius
+
+		UAS_FCU(m_uas)->send_message_ignore_drop(of);
+	}
+
+	/**
+	 * @brief Send simulated received RAW values of the RC channels to the FCU.
+	 * Message specification: @p https://pixhawk.ethz.ch/mavlink/#HIL_RC_INPUTS_RAW
+	 */
+	void rcin_raw_cb(const mavros_msgs::RCIn::ConstPtr &req) {
+		mavlink::common::msg::HIL_RC_INPUTS_RAW rcin;
+
+		constexpr size_t MAX_CHANCNT = 12;
+		lock_guard lock(mutex);
+		size_t channels_count = req->channels.size();
+
+		if (channels_count > MAX_CHANCNT) {
+			ROS_WARN_THROTTLE_NAMED(60, "hil_rc",
+						"FCU receives %lu RC channels, but HIL_RC_INPUTS_RAW can store %zu",
+						req->channels.size(), MAX_CHANCNT);
+
+			channels_count = MAX_CHANCNT;
+		}
+
+		rcin.time_usec = req->header.stamp.toNSec() / 100000;
+		// switch works as start point selector.
+		switch (channels_count) {
+		// [[[cog:
+		// for i in range(12, 0, -1):
+		//     cog.outl("case %2d: rcin.chan%d_raw \t= req->channels[%2d];" % (i, i, i-1))
+		// ]]]
+		case 12: rcin.chan12_raw	= req->channels[11];
+		case 11: rcin.chan11_raw	= req->channels[10];
+		case 10: rcin.chan10_raw	= req->channels[ 9];
+		case  9: rcin.chan9_raw		= req->channels[ 8];
+		case  8: rcin.chan8_raw		= req->channels[ 7];
+		case  7: rcin.chan7_raw		= req->channels[ 6];
+		case  6: rcin.chan6_raw		= req->channels[ 5];
+		case  5: rcin.chan5_raw		= req->channels[ 4];
+		case  4: rcin.chan4_raw		= req->channels[ 3];
+		case  3: rcin.chan3_raw		= req->channels[ 2];
+		case  2: rcin.chan2_raw		= req->channels[ 1];
+		case  1: rcin.chan1_raw		= req->channels[ 0];
```

Comment:
I mean that if unused channel should be set to UINT16_MAX that approach is not good enough.
If 0 - is ok, then that solution is ok.